olp-cpp-sdk-dataservice-read/src/repositories/PrefetchTilesRepository.cpp

```diff
@@ -48,117 +48,141 @@ constexpr std::uint32_t kMaxQuadTreeIndexDepth = 4u;
 
 using namespace olp::client;
 
-SubQuadsRequest PrefetchTilesRepository::EffectiveTileKeys(
-    const std::vector<geo::TileKey>& tile_keys, unsigned int min_level,
-    unsigned int max_level) {
-  SubQuadsRequest ret;
-  for (auto tile_key : tile_keys) {
-    auto child_tiles = EffectiveTileKeys(tile_key, min_level, max_level, true);
-    for (auto tile : child_tiles) {
-      // check if child already exist, if so, use the greater depth
-      auto old_child = ret.find(tile.first);
-      if (old_child != ret.end()) {
-        ret[tile.first] = std::max((*old_child).second, tile.second);
-      } else {
-        ret[tile.first] = tile.second;
-      }
+std::vector<geo::TileKey> PrefetchTilesRepository::GetChilds(
+    const geo::TileKey& tile_key) {
+  if (!tile_key.IsValid()) {
+    return {};
+  }
+  std::vector<geo::TileKey> ret;
+  for (std::uint8_t index = 0; index < kMaxQuadTreeIndexDepth; ++index) {
+    auto tile = tile_key.GetChild(index);
+    if (tile_key.IsValid()) {
+      ret.emplace_back(std::move(tile));
     }
   }
-
   return ret;
 }
```

Comment:
In the end, there is no need to construct a vector, all function can be replaced with 4 lines:
vector.emplace_back(tile_key.GetChild(0));
vector.emplace_back(tile_key.GetChild(1));
vector.emplace_back(tile_key.GetChild(2));
vector.emplace_back(tile_key.GetChild(3));